Sources/Afluent/SequenceOperators/RetrySequence.swift

```diff
@@ -71,6 +71,77 @@ extension AsyncSequences {
 
         public nonisolated func makeAsyncIterator() -> Retry<Upstream, Strategy> { self }
     }
+    
+    public final actor RetryOn<Upstream: AsyncSequence & Sendable, Failure: Error & Equatable, Strategy: RetryStrategy>: AsyncSequence, AsyncIteratorProtocol, Sendable where Upstream.Element: Sendable {
```

Comment:
Just a thought: the "retry on error" variants are so _so_ similar to the normal "retry" operators, I wonder if we can use an optional closure for the error matching logic? Instead of requiring the additional generic?

Could use an enum or something to give it some structure:
```swift
enum OnError {
  case matches((any Error) -> Bool)
  case `any`

  func matches(_ error: any Error) -> Bool {
    guard case .matches(let matcher) = self else { return true }
    return matcher(error)
  } 
}
```